### PR TITLE
Support reading default application credentials for analytics uploading

### LINKED ISSUE #68: Analytics using default credentials if given
Some small change needs to be make to use default credentials (either located at `GOOGLE_APPLICATION_CREDENTIALS` or from the GKE/GCE metadata server).

An open question is whether this should be prioritized over a given credentials json.

### PR DESCRIPTION
**Pull request by rockspore**
Fix #68 and #73

### CODE REVIEW

File: server/config.go [hunk 1]
```diff
@@ -231,8 +233,9 @@ func (c *Config) Load(configFile, policySecretPath, analyticsSecretPath string)
 			}
 		}
 
-		// attempts to load the service account credentials if the credentials are not set yet
-		if analyticsSecretPath != "" && c.Analytics.CredentialsJSON == nil {
+		// attempts to load the service account credentials if a path is given
+		if analyticsSecretPath != "" {
+			log.Debugf("using analytics service account credentials from the given path")
```
Comment (theganyo) [inline server/config.go:238]: print the path instead of saying "given path"?

File: server/config.go [hunk 2]
```diff
@@ -265,8 +268,17 @@ func (c *Config) Validate() error {
 	if c.Tenant.RemoteServiceAPI == "" {
 		errs = multierror.Append(errs, fmt.Errorf("tenant.remote_service_api is required"))
 	}
-	if c.Tenant.InternalAPI == "" && c.Analytics.FluentdEndpoint == "" && c.Analytics.CredentialsJSON == nil {
-		errs = multierror.Append(errs, fmt.Errorf("tenant.internal_api or tenant.analytics.fluentd_endpoint is required if no service account"))
+	if c.Analytics.CredentialsJSON == nil {
+		if c.Tenant.InternalAPI == "" && c.Analytics.FluentdEndpoint == "" {
+			errs = multierror.Append(errs, fmt.Errorf("tenant.internal_api or tenant.analytics.fluentd_endpoint is required if no service account"))
+		}
+		// if c.Tenant.InternalAPI != "" && c.Analytics.FluentdEndpoint != "" {
```
Comment (theganyo) [inline server/config.go:275]: remove dead code?

File: server/handler.go [hunk 3]
```diff
@@ -236,15 +246,30 @@ func roundTripperWithPrometheus(config *Config, api string, rt http.RoundTripper
...
+	if jsonData != nil { // always uses given credentials
+		cred, err := google.CredentialsFromJSON(ctx, jsonData, scope)
+		if err != nil { // stop proceeding if given credentials are in bad format
+			return nil, err
+		}
+		log.Debugf("analytics http client using provided analytics service account credentials")
+		client = oauth2.NewClient(ctx, cred.TokenSource)
+	}
+
+	if client == nil {
```
Comment (theganyo) [inline server/handler.go:264]: Is this just an else clause?

Comment (rockspore) [reply server/handler.go:264]: Yes. Thanks!

**Review by theganyo — APPROVED** [review]

**Pull request merged**